pkg/test/e2e/ccm-migration/ccm_migration_suite_test.go

```diff
@@ -26,39 +26,50 @@ import (
 }
 
 var options = testOptions{
-	kubernetesVersion: *semver.NewSemverOrDie("v1.20.14"),
+	kubernetesVersion: *semver.NewSemverOrDie("v1.21.7"),
 }
 
 func init() {
-	flag.StringVar(&options.datacenter, "datacenter", "byo-kubernetes", "Name of the datacenter used by the user clusters created for the test.")
-	flag.StringVar(&options.userClusterName, "user-cluster-name", "", "name of the user cluster to be created")
 	flag.Var(&options.kubernetesVersion, "kubernetes-version", "Kubernetes version for the user cluster")
 	flag.BoolVar(&options.debugLog, "debug-log", false, "Activate debug logs.")
 	flag.BoolVar(&options.skipCleanup, "skip-cleanup", false, "Skip clean-up of resources.")
 
-	flag.StringVar(&options.osCredentials.authURL, "openstack-auth-url", "", "openstack auth url")
-	flag.StringVar(&options.osCredentials.username, "openstack-username", "", "openstack username")
-	flag.StringVar(&options.osCredentials.password, "openstack-password", "", "openstack password")
-	flag.StringVar(&options.osCredentials.tenant, "openstack-tenant", "", "openstack tenant")
-	flag.StringVar(&options.osCredentials.domain, "openstack-domain", "", "openstack domain")
-	flag.StringVar(&options.osCredentials.region, "openstack-region", "", "openstack region")
-	flag.StringVar(&options.osCredentials.floatingIPPool, "openstack-floating-ip-pool", "", "openstack floating ip pool")
-	flag.StringVar(&options.osCredentials.network, "openstack-network", "", "openstack network")
+	flag.StringVar(&options.osSeedDatacenter, "openstack-seed-datacenter", "", "openstack datacenter")
+	flag.StringVar(&options.vsphereSeedDatacenter, "vsphere-seed-datacenter", "", "vsphere seed datacenter")
+
+	flag.StringVar(&options.osCredentials.AuthURL, "openstack-auth-url", "", "openstack auth url")
+	flag.StringVar(&options.osCredentials.Username, "openstack-username", "", "openstack username")
+	flag.StringVar(&options.osCredentials.Password, "openstack-password", "", "openstack password")
+	flag.StringVar(&options.osCredentials.Tenant, "openstack-tenant", "", "openstack tenant")
+	flag.StringVar(&options.osCredentials.Domain, "openstack-domain", "", "openstack domain")
+	flag.StringVar(&options.osCredentials.Region, "openstack-region", "", "openstack region")
+	flag.StringVar(&options.osCredentials.FloatingIPPool, "openstack-floating-ip-pool", "", "openstack floating ip pool")
+	flag.StringVar(&options.osCredentials.Network, "openstack-network", "", "openstack network")
+
+	flag.StringVar(&options.vSphereCredentials.AuthURL, "vsphere-auth-url", "", "vsphere auth-url")
+	flag.StringVar(&options.vSphereCredentials.Username, "vsphere-username", "", "vsphere username")
+	flag.StringVar(&options.vSphereCredentials.Password, "vsphere-password", "", "vsphere password")
+	flag.StringVar(&options.vSphereCredentials.Datacenter, "vsphere-datacenter", "", "vsphere datacenter")
+	flag.StringVar(&options.vSphereCredentials.Cluster, "vsphere-cluster", "", "vsphere cluster")
```

Comment:
Is it possible that we handle credentials in some other way, e.g. via environment variables? Having credentials in the plain text provided via command-line flags can very easily lead to leaking them.

Reply:
This is the same approach that we use in the conformance tests. We set env variables via prow to the container and then fetch and provide them as command-line flags to the test. Do you suggest retrieving the env vars directly from the test executable?